src/controllers/participant.controller.ts

```diff
@@ -1,264 +1,256 @@
-            { acceptanceStatus: StatusEnum.REJECTED },
-            { phone }
-        );
-        await dbAddNewLog({
-            adminPhone: req.user?.phone as string,
-            adminId: req.user?._id as string,
-            action: "update",
-            participantId: updatedParticipant._id as string,
-        });
-        res.status(200).json({ status: "success", data: updatedParticipant });
-    } catch (error: ErrorWithStatusCode | any) {
-        res.status(error.statusCode || 500).json({
-            status: "failure",
-            data: error.message,
-        });
-    }
+	/**
+	 * #swagger.tags = ['Participants']
+	 * #swagger.description = 'Endpoint to get reject a participant from database'
+	 */
+	try {
+		const { phone }: { phone: string } = req.body;
+		const originalParticipant = await dbGetParticipantByPhone(phone);
+		const updatedParticipant = await dbUpdateParticipant({ acceptanceStatus: StatusEnum.REJECTED }, { phone });
+		const diff = diffObjects(originalParticipant, updatedParticipant);
+		await dbAddNewParticipantLog({
+			initiator: req.user?._id as string,
+			action: "update",
+			target: updatedParticipant._id as string,
+			previousState: diff.OLD as Partial<ParticipantType>,
+			newState: diff.NEW as Partial<ParticipantType>,
+		});
+		res.status(200).json({ status: "success", data: updatedParticipant });
+	} catch (error: ErrorWithStatusCode | any) {
+		res.status(error.statusCode || 500).json({
+			status: "failure",
+			data: error.message,
+		});
+	}
 }
 
 export async function emailParticipantByPhone(req: Request, res: Response) {
-    /**
-     * #swagger.tags = ['Participants']
-     * #swagger.description = 'Endpoint to get send an email to a participant from database'
-     */
-    try {
-        const { phone }: { phone: string } = req.body;
-        //TODO: send email using nodemailer here
-        const updatedParticipant = await dbUpdateParticipant(
-            { acceptanceStatus: StatusEnum.EMAILED },
-            { phone }
-        );
-        await dbAddNewLog({
-            adminPhone: req.user?.phone as string,
-            adminId: req.user?._id as string,
-            action: "update",
-            participantId: updatedParticipant._id as string,
-        });
-        res.status(200).json({ status: "success", data: updatedParticipant });
-    } catch (error: ErrorWithStatusCode | any) {
-        res.status(error.statusCode || 500).json({
-            status: "failure",
-            data: error.message,
-        });
-    }
+	/**
+	 * #swagger.tags = ['Participants']
+	 * #swagger.description = 'Endpoint to get send an email to a participant from database'
+	 */
+	try {
+		const { phone }: { phone: string } = req.body;
+		//TODO: send email using nodemailer here
+		const originalParticipant = await dbGetParticipantByPhone(phone);
+		const updatedParticipant = await dbUpdateParticipant({ acceptanceStatus: StatusEnum.EMAILED }, { phone });
```

Comment:
You don't need 2 database calls for this. There's a function in MongoDB that updates a document and returns it, and can be controlled to return the original. Using this, you will be able to update and get the original in one step, and you already have the updated state passed into the function.